packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -977,6 +1094,12 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         }
     };
 
+    private doHandleContextMenuContributed(payload: ContextMenuContributedSignalPayload) {
+        if (payload.getOutputDescriptor().id === this.props.outputDescriptor.id) {
+            this.setState({ ctxMenu: payload.getMenu() });
```

Comment:
Hmm good point - Maybe i should change it so that the signals can pass menu items rather an entire context menu - That way it will have a generic menu with the ability to add as many menu items as needed from multiple sources